pkg/apis/sources/v1alpha1/containersource_types.go

```diff
@@ -52,22 +52,34 @@ var (
 
 // ContainerSourceSpec defines the desired state of ContainerSource
 type ContainerSourceSpec struct {
+	// Template describes the pods that will be created
+	// +optional
+	Template *corev1.PodTemplateSpec `json:"template,omitempty"`
+
 	// Image is the image to run inside of the container.
+	// This field is to be deprecated. Use `Template` instead.
+	// When `Template` is set, this field is ignored.
 	// +kubebuilder:validation:MinLength=1
 	Image string `json:"image,omitempty"`
```

Comment:
Can you change to:

```
DeprecatedImage string `json:"image,omitempty"`
DeprecatedArgs []string `json:"args,omitempty"`
DeprecatedEnv []corev1.EnvVar `json:"env,omitempty" patchStrategy:"merge" patchMergeKey:"name"`
DeprecatedServiceAccountName string `json:"serviceAccountName,omitempty"`
```

Reply:
Good call. Done in https://github.com/knative/eventing/pull/1321/commits/c6bb6220128fc77d0d201c6013039d37d2ae5037.